sklearn/cluster/_hdbscan/_linkage.pyx

```diff
@@ -13,9 +13,9 @@ from ...cluster._hierarchical_fast cimport UnionFind
 from ...cluster._hdbscan._tree cimport HIERARCHY_t
 from ...cluster._hdbscan._tree import HIERARCHY_dtype
 from ...utils._typedefs cimport intp_t, float64_t, int64_t, uint8_t
+from numpy cimport PyArrayObject
```

Comment:
Do you think occurrences of `PyArrayObject` be replaced by `cnp.PyArrayObject` instead?

Reply:
I prefer import types directly (e.g. `HIERARCHY_t`) but I don't have a huge preference. I have updated them now :)

Reply:
So do I, I think that we tend to import NumPy's symbol using `cnp.` everytime (hence my question). But that's minor as a comment.